src/main/java/net/smartcosmos/edge/things/service/GetThingEdgeServiceDefault.java

```diff
@@ -194,18 +240,29 @@ public GetThingEdgeServiceDefault(
     }
 
     private String getByTypeLogMessage(String type, SmartCosmosUser user, String message) {
+
         return String.format("Read request for Thing with type '%s' by user '%s' failed: %s",
                              type,
                              user,
                              message);
     }
 
     private String getByTypeAndUrnLogMessage(String type, String urn, SmartCosmosUser user, String message) {
+
         return String.format("Read request for Thing with type '%s' and urn '%s' by user '%s' failed: %s",
                              type,
                              urn,
                              user,
                              message);
     }
 
+    private String getByTypeAndUrnsLogMessage(String type, Map<String, Set<String>> urns, SmartCosmosUser user, String message) {
```

Comment:
I still don't like this approach.  Re-using the exact same message for what essentially appears to be different causes will prompt confusion in the called when the failure happens.  In this case the situation can be 

1. A failure getting the thing(s) being requested, or 
2. a failure getting metadata for a thing already retrieved.

These are not the same.  If we provide unique messages we give the user a chance of identifying the underlying issue with the service that is causing the problem.